### PR TITLE
Add expression functions available in layout legend item

### PR DESCRIPTION
**Pull request by DelazJ**
as part of #4186

### CODE REVIEW

File: docs/user_manual/working_with_vector/expression.rst [hunk 1]
```diff
@@ -1287,6 +1287,10 @@ To use these functions in an expression, they should be preceded by @ character
    "symbol_angle", "Returns the angle of the symbol used to render the feature
    (valid for marker symbols only)"
    "symbol_color", "Returns the color of the symbol used to render the feature"
+   "symbol_count", "Returns the total number of features represented by the symbol (in layout legend)"
+   "symbol_id", "Returns the Internal ID of the symbol (in layout legend)"
+   "symbol_label", "Returns the label for the symbol (either a user defined
+   label or the default autogenerated label - in layout legend)"
```
Comment (havatv) [inline docs/user_manual/working_with_vector/expression.rst:1293]:
```suggestion
   label or the default autogenerated label - in the layout legend)"
```

File: docs/user_manual/working_with_vector/expression.rst [hunk 2]
```diff
@@ -1287,6 +1287,10 @@ To use these functions in an expression, they should be preceded by @ character
    "symbol_angle", "Returns the angle of the symbol used to render the feature
    (valid for marker symbols only)"
    "symbol_color", "Returns the color of the symbol used to render the feature"
+   "symbol_count", "Returns the total number of features represented by the symbol (in layout legend)"
```
Comment (havatv) [inline docs/user_manual/working_with_vector/expression.rst:1290]:
```suggestion
   "symbol_count", "Returns the number of features represented by the symbol (in the layout legend)"
```

File: docs/user_manual/working_with_vector/expression.rst [hunk 3]
```diff
@@ -1287,6 +1287,10 @@ To use these functions in an expression, they should be preceded by @ character
...
+   "symbol_id", "Returns the Internal ID of the symbol (in layout legend)"
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/expression.rst:1291]:
```suggestion
   "symbol_id", "Returns the Internal ID of the symbol (in the layout legend)"
```

**Comment by DelazJ** [comment]
manual backport at https://github.com/qgis/QGIS-Documentation/commit/70c053843c50fc5585eea90c5d78d3f1e97cfe34

**Pull request merged**